server/cluster/coordinator.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package cluster
+
+import (
+	"context"
+	"sync"
+
+	"github.com/CeresDB/ceresdbproto/pkg/clusterpb"
+	"github.com/pkg/errors"
+)
+
+type coordinator struct {
+	sync.Mutex
+	cluster *Cluster
+}
+
+func (c *coordinator) Run(ctx context.Context) error {
+	c.Lock()
+	defer c.Unlock()
+	if err := c.scatterShard(ctx); err != nil {
+		return errors.Wrap(err, "coordinator Run")
+	}
+	return nil
+}
+
+// todo: consider ReplicationFactor
+func (c *coordinator) scatterShard(ctx context.Context) error {
+	if !(int(c.cluster.metaData.cluster.MinNodeCount) >= len(c.cluster.metaData.nodeMap) &&
+		c.cluster.metaData.clusterTopology.State == clusterpb.ClusterTopology_EMPTY) {
+		return nil
+	}
+
+	c.cluster.Lock()
+	defer c.cluster.Unlock()
+
+	shardTotal := int(c.cluster.metaData.cluster.ShardTotal)
+	minNodeCount := int(c.cluster.metaData.cluster.MinNodeCount)
+	perNodeShardCount := shardTotal / minNodeCount
+	shards := make([]*clusterpb.Shard, shardTotal)
+	nodeIndex := 0
+	var nodeList []*clusterpb.Node
+	for _, v := range c.cluster.metaData.nodeMap {
+		nodeList = append(nodeList, v)
+	}
+
+	for i := 0; i < minNodeCount; i++ {
+		for j := 0; j < perNodeShardCount; j++ {
```

Comment:
What if the perNodeShardCount == 0?

Reply:
Add parameter checks when creating clusters.